src/DependencyInjection/M6WebGuzzleHttpExtension.php

```diff
@@ -54,14 +54,19 @@ protected function loadClient(ContainerBuilder $container, $clientId, array $con
             $config['allow_redirects'] = false;
         }
 
-        $this->setGuzzleProxyHandler($container, $clientId, $config);
-
         $handlerStackDefinition = new Definition('%m6web_guzzlehttp.guzzle.handlerstack.class%');
         $handlerStackDefinition->setPublic(true);
         $handlerStackDefinition->setFactory(['%m6web_guzzlehttp.guzzle.handlerstack.class%', 'create']);
-        $handlerStackDefinition->setArguments([new Reference('m6web_guzzlehttp.guzzle.proxyhandler_'.$clientId)]);
 
-        $container->setDefinition('m6web_guzzlehttp.guzzle.handlerstack.'.$clientId, $handlerStackDefinition);
+        if (!isset($config['stack_handler'])) {
+            $this->setGuzzleProxyHandler($container, $clientId, $config);
+
+            $handlerStackDefinition->setArguments([new Reference('m6web_guzzlehttp.guzzle.proxyhandler_' . $clientId)]);
+        } else {
+            $handlerStackDefinition->setArguments([new Reference($config['stack_handler'])]);
+        }
+
+        $container->setDefinition('m6web_guzzlehttp.guzzle.handlerstack.' . $clientId, $handlerStackDefinition);
```

Comment:
`m6web_guzzlehttp.guzzle.handlerstack.$clientId`

vs

`m6web_guzzlehttp.guzzle.proxyhandler_$clientId`

it's sad that we're not consistent in our service id naming convention (point vs underscore) 😞